pkg/ui/console/client.go

```diff
@@ -407,28 +407,60 @@ func (c Client) UpgradeProviders(ctx context.Context, args []string) error {
 	return nil
 }
 
-//nolint:dupl
 func (c Client) DowngradeProviders(ctx context.Context, args []string) error {
 	providers, err := c.getRequiredProviders(args)
 	if err != nil {
 		return err
 	}
+
+	// download requested versions
 	if err := c.DownloadProviders(ctx); err != nil {
 		return err
 	}
+
+	ui.ColorizedOutput(ui.ColorProgress, "Will upgrade providers first, before downgrade... %s\n\n", strings.Join(providers.Names(), ", "))
+
+	provVersions := make(map[string]string, len(providers))
+	for _, p := range providers {
+		provVersions[p.Name] = p.Version
+		p.Version = migrator.Latest
+	}
+
+	newCfg := *c.cfg
+	newCfg.CloudQuery.Providers = providers
+	conWithLatest, err := CreateClientFromConfig(ctx, &newCfg)
```

Comment:
wouldn't conWithLatest (version X > Y) be able to downgrade Y -> Z  (Y > Z) I don't think we need to do an upgrade with the cqWithLatest.

Reply:
Fixed